### ISSUE TITLE
SmtpBuilder: does not support non SSL/TLS connection?

### ISSUE THREAD

**Issue by nextfool** [nopSolutions/nopCommerce#5574]
nopCommerce version: 4.40.2

Steps to reproduce the problem: 
``` 
    try
            {
                await client.ConnectAsync(
                    emailAccount.Host,
                    emailAccount.Port,
                    emailAccount.EnableSsl ? SecureSocketOptions.SslOnConnect : SecureSocketOptions.**StartTlsWhenAvailable**);

                if (emailAccount.UseDefaultCredentials)
                {
                    await client.AuthenticateAsync(CredentialCache.DefaultNetworkCredentials);
                } 
                else if (!string.IsNullOrWhiteSpace(emailAccount.Username))
                {
                    await client.AuthenticateAsync(new NetworkCredential(emailAccount.Username, emailAccount.Password));
                }

                return client;
            }
```

So  in the email account settings if we  left "Enable SSL" unchecked, as you can see it still fallback to TLS connection?
Isn't it suppose to be SecureSocketOptions.None here if we don't want any authentication here?

**Comment by skoshelev** [comment]
Hi @nextfool.

We specifically use the StartTlsWhenAvailable flag to protect user data. If TLS is not available on the server, then the behavior of this flag is no different from SecureSocketOptions.None, so we see no need to create a security problem by directly using the SecureSocketOptions.None flag

**Comment by nextfool** [comment]
@skoshelev Sorry I didn't follow this session in time.  I my case, I am using the Alibaba "mail direct" as my smtp service provider, and it only work when I set the flag to "SecureSocketOptions.None". Otherwise an exception occurs complaining about "TLS" connection issue. At the moment I just modified the code to fix my problem, but can you maybe check this out?

**Comment by skoshelev** [comment]
Hi @exileDev, please check

**Comment by exileDev** [comment]
We'll think about using additional values to specifying the SSL/TLS encryption.

**Comment by exileDev** [comment]
@nextfool Hi! I understand this topic is quite old, but it would be great if you'll provide us with some details. According to DirectMail's documentation:

> Port 25 is currently disabled to maintain ECS security. If your sending application is deployed on Alibaba Cloud ECS, we recommend that you use port 80 when SSL is not checked, and use port 465 when SSL is checked. Besides, we support explict ssl/tls functionality on port 25 or 80 with “STARTTLS” smtp cmd.

So you can try to configure SMTP on 80/465 ports. If you don't want to use TLS, you can unset the Use default credentials option and leave the username/password is empty.

**Comment by exileDev** [comment]
We'll reopen this issue if we find out more.
Closed #5574
